boa_engine/src/value/into_js_value.rs

```diff
@@ -0,0 +1,127 @@
+use super::JsValue;
+use crate::{
+    Context,
+    builtins::Array
+};
+
+pub trait IntoJsValue {
+    fn into_js_value(self, context: &mut Context) -> JsValue;
+}
+
+impl IntoJsValue for bool {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl IntoJsValue for f64 {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl IntoJsValue for f32 {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl IntoJsValue for i128 {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl IntoJsValue for i64 {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl IntoJsValue for i32 {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl IntoJsValue for i16 {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl IntoJsValue for i8 {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl<T: IntoJsValue> IntoJsValue for Option<T> {
+    fn into_js_value(self, context: &mut Context) -> JsValue {
+        match self {
+            Some(value) => value.into_js_value(context),
+            None => JsValue::Null.into()
+        }
+    }
+}
+
+impl IntoJsValue for String {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl IntoJsValue for u128 {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl IntoJsValue for u64 {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl IntoJsValue for u32 {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl IntoJsValue for u16 {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+impl IntoJsValue for u8 {
+    fn into_js_value(self, _: &mut Context) -> JsValue {
+        self.into()
+    }
+}
+
+// TODO consider that each type might need its own explicit impl for Vec
+// TODO the derive attribute might need to be used in that case
+impl<T: IntoJsValue> IntoJsValue for Vec<T> {
+    fn into_js_value(self, context: &mut Context) -> JsValue {
+        let js_values = self.into_iter().map(|item| item.into_js_value(context)).collect::<Vec<JsValue>>();
+
+        Array::create_array_from_list(
+            js_values,
+            context
+        ).into()
+    }
+}
```

Comment:
Exposing `Array` is pretty unnecessary, since there's already an API that allows to convert any `Iterator<Item=JsValue>` into an array.

https://github.com/boa-dev/boa/blob/main/boa_engine/src/object/jsarray.rs

Reply:
Yeah I'm a bit confused on how to do this the right way, can you give me a little code hint?

Reply:
Oh I think I see, from_iter would give me a JsArray and then I could get it to a JsValue from there, let me work on that

Reply:
This has been addressed